templates/rate-plan-detail.html.twig

```diff
@@ -24,44 +28,108 @@
+ * - ratecard_rates:             The RATECARD rates.
+ *   - id:                          The unique ID of the rate plan rate.
+ *   - startUnit:                   The number the rate starts at.
+ *   - endUnit:                     The rate is good until this nuber of units.
+ *   - rate:                        The rate that is applies within this range.
+ * - revshare_rates:             The REVSHARE rates.
+ *   - id:                          The unique ID of the rate plan rate.
+ *   - startUnit:                   The number the rate starts at.
+ *   - endUnit:                     The rate is good until this nuber of units.
+ *   - revshare:                    The revenue share percentage.
+ * - entity:                     The rate plan entity.
  */
 #}
 
-{% set detail_fields = [
-  {
-    'label': "Renewal Period",
-    'value': detail.duration ~ ' ' ~ detail.durationType|lower
-  },
-  {
-    'label': "Operator",
-    'value': detail.organization.description,
-  },
-  {
-    'label': "Country",
-    'value': detail.organization.country,
-  },
-  {
-    'label': "Currency",
-    'value': detail.currency.name,
-  },
-  {
-    'label': "Pricing Type",
-    'value': detail.revenueType|lower|capitalize
-  },
-  {
-    'label': "Rev Share %",
-    'value': detail.ratePlanRates.RevShare,
-    'suffix': '%'
-  },
+{% set title = detail.type == 'REVSHARE_RATECARD' ? "Rate card & revenue" : detail.type == 'REVSHARE' ? "Revenue" : "Rate card" %}
 
-]%}
+<div class="rate-plan-detail">
+  <h2 class="rate-plan-detail__title">{{ title|t }}</h2>
+  <div class="field field--inline rate-plan-detail__metering-type">
+    <div class="field__label">{{ "Rate card is based on"|t }}</div>
+    <div class="field__item">{{ detail.meteringType }}</div>
+  </div>
+  <div class="field field--inline rate-plan-detail__metering-basis">
+    <div class="field__label">{{ "Volume aggregation basis"|t }}</div>
+    <div class="field__item">{{ detail.duration }} {{ (detail.duration == 1 ? detail.durationType : detail.durationType ~ 's')|lower|t }}</div>
+  </div>
 
-<table class="rate-plan-detail">
-{% for detail_field in detail_fields %}
-  {% if detail_field.value %}
-  <tr class="field--label-inline">
-    <td><label class="field__label">{{ detail_field.label }}</label></td> <td><span>{{ detail_field.value }} {{ (detail_field.suffix) ? '%' }}</span></td>
-  </tr>
+  {% if entity.freemiumUnit or entity.freemiumDuration %}
+    <div class="field field--inline rate-plan-detail__freemium">
+      <div class="field__label">{{ "Free quantity"|t }}</div>
+      <div class="field__item">
+        {% if entity.freemiumUnit and entity.freemiumDuration and entity.freemiumDurationType %}
+          {{ "Up to"|t }} {{ entity.freemiumUnit }} {{ (entity.freemiumUnit == 1 ? "transaction for" : "transactions for")|t }} {{ entity.freemiumDuration }} {{ (entity.freemiumDuration == 1 ? entity.freemiumDurationType : entity.freemiumDurationType ~ 's')|lower|t }}.
```

Comment:
These strings are really, really messy and difficult to grok.  Typically you are supposed to use `{% trans %}` with `{% plural %}` but having to run filters on this stuff too, make it even more complex than it already is, because you [can't even run filters](https://www.drupal.org/project/drupal/issues/2610436) inside trans. It would help if you at least did the `|lower` transform in the module code, and honestly the strings themselves, in the module code.

Reply:
I moved this logic to the formatter. It's probably better since the real values here come from the rate plan entity and not the details property.